llms/mlx_lm/lora.py

```diff
@@ -229,11 +254,38 @@ def train_model(
     )
 
     model.train()
-    opt = optim.Adam(
-        learning_rate=(
-            build_schedule(args.lr_schedule) if args.lr_schedule else args.learning_rate
+
+    # Initialize the selected optimizer
+    lr = build_schedule(args.lr_schedule) if args.lr_schedule else args.learning_rate
+    
+    optimizer_name = args.optimizer.lower()
+    optimizer_config = args.optimizer_config.get(optimizer_name, {})
+    
+    if optimizer_name == "adam":
```

Comment:
Change the switch to set `opt_class = optim.Adam` (e.g. just the class).

Then make the `opt` from the config and class like so:

```
opt = opt_class(**optimizer_config)
```

Also in the config only set parameters which don't have a default for the optimizer. Otherwise let's just inherit the defaults rather than duplicating them here.